### PR TITLE
Fix issues from #1047

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
I didn't put the 'waiting' logic for the server-side nav case in the right location.  This PR fixes that.

One of the things from the ADR was wanted to make sure people are using 'beforeNav' and 'safeWithNav' only when 'href' is set.  This PR also fixes that by changing the Button props to be a disjoint union.

Issue: WEB-2990

## Test plan:
- yarn jest

### CODE REVIEW

**Review by jeresig — APPROVED** [review]

File: packages/wonder-blocks-button/components/__tests__/button.flowtest.js
Comment (aag) [inline packages/wonder-blocks-button/components/__tests__/button.flowtest.js:12]: Maybe add one more test to verify that `href` can be used by itself?

```
// It's okay to use href by itself
<Button href="/foo">
    Hello, world!
</Button>;
```

**Review by aag — APPROVED** [review]
Looks good to me. I suggested 1 additional test, but I'll leave it up to you to add it or not.

File: packages/wonder-blocks-button/components/button.js
Comment (somewhatabstract) [inline packages/wonder-blocks-button/components/button.js:165]: Thanks for doing this. Be prepared for webapp issues though if folks spread these props for some reason. Our version of flow in webapp is not good to spread unions.

File: packages/wonder-blocks-core/components/clickable-behavior.js
Comment (somewhatabstract) [inline packages/wonder-blocks-core/components/clickable-behavior.js:338]: This terminology is a bit confusing (at least for me) as I immediately think of "server-side rendering". Perhaps consider "full page load navigation"?

**Review by somewhatabstract — APPROVED** [review]
LGTM

Some inline thoughts. Nothing serious.

File: packages/wonder-blocks-button/components/button.js
Comment (kevinbarabash) [reply packages/wonder-blocks-button/components/button.js:165]: Thanks for the heads up.

File: packages/wonder-blocks-core/components/clickable-behavior.js
Comment (kevinbarabash) [reply packages/wonder-blocks-core/components/clickable-behavior.js:338]: "full page load navigation" is better.  I'll update this before landing.

**Pull request merged**
